johnsonandjohnson/Bodiless-JS
# fix(starter-kit): Starter kit cleanup

## Changes
Bring over components from test site:

- src/components/Editors
  - only change was removing block-quotes and strike-thru as the are not commonly used in site builds
- src/components/Flowcontainer
  - removed Accordions & Contentful variations
- src/components/Images
  - no change from test-site
- src/components/Layout
  - no change from test-site PR: https://github.com/johnsonandjohnson/Bodiless-JS/pull/455
- src/components/Menu 
   - from test-site PR: https://github.com/johnsonandjohnson/Bodiless-JS/pull/455 plus modified some colors and names of tokens to be generic 
- src/components/Page
  - no change from test-site
- src/components/Tout
  - no change from test-site
- src/components/Elements.token.ts
  - brought over from test-site minus strikethru 

Pages
- Homepage redone as flow-container
    - Added meta data
    - Added Data
- 404 redone as flow-container. 
    - Added meta data 
    - Added Data
 

Templates
 - Updated to use flow-container

Gatsby config & Package Files
- Added default support of gatsby-plugin-canonical-urls & gatsby-plugin-sitemap

Tailwind Config 
- Brought over nice commenting from test-site and removed custom background color for burger-menu

Env file
- Changed port to use 8001.

Extra items from documentation/discussion that affect both Test & Starter Kit:
- Removed dependency on siteMetadata and made title & logo editable.
- SiteURL remains but moved to ENV site and shared with Canonical URLS.  So its set in one place instead of two.
- Switched menu, footer, & logo over to use withNodeKey to encourage that pattern.
- small refactor of GTM data layer to its own file 
- Updated favicon to use generic name 'favicon.png' and added manifest plugin to gatsby.config to not use settings from gatsby-theme-bodiless.
 - fixed linting issues


Update documentation:
- /Development/Guides/SiteBuildBasics?id=site-level-data
- /Development/Guides/BuildingSites/BuildingOverview
- /Development/Guides/BuildingSites/ComponentsStarterKit
- README

## Test Instructions

Recommend in both cases for testing:  to create sites outside of monorepo and bodiless pack gatsby-theme-bodiless:

- Test the Example Test Site
   - Logo is now editable image and should appear in both desktop and mobile version.
   - Site title and Site copyright is editable (only Site Copyright was editable before)
   - Confirm Logo, Menu, Site Title, Site Copyright are all saving in src/data/site
- Test the Starter Test Site
   - Header (Test that image logo is editable image, the image link is not editable and fixed to '/'
   - Menu (Menu/burger Menu behaves like test site)
   - Footer (Site title and Site copyright is editable)
   - Pages Exist
      - Home page with flowcontainer and components on it.
      - 404 page with flowcontainer and default 404 copy as previously defined.
   - Canonical URLS are populating on static.
   - Sitemap.xml is populating on static.
   - Flowcontainer works as expected for all components.
   - GTM is on and working correctly on static.
   - Meta is working with home & 404 pages
   - Favicon should be set.
 - Confirm starter backend runs on 8001

## Related Issues
